test/unit/3p/test-3p-frame.js

```diff
@@ -42,8 +43,11 @@ describes.realWin('3p-frame', {amp: true}, (env) => {
     document = window.document;
   });
 
-  function mockMode(mode) {
-    env.sandbox.stub(window.parent, '__AMP_MODE').value(mode);
+  function mockMode(options) {
```

Comment:
Since this shift is going to be happening in a lot of places, it might be useful (possibly not in this PR) to provide it as a helper somewhere in `testing/` a la

```javascript
  function mockMode(options) {
    options.entries.forEach(([prop, val]) => env.sandbox.stub(mode, prop).returns(val));
  }   
```

@rsimha for thoughts on if/where this should live

Reply:
Good idea. As it turns out, @rileyajones is already working on extracting commonly used testing functionality into a new file `testing/helpers.js`. The in-flight PR is #35531 (see https://github.com/ampproject/amphtml/pull/35531/files#diff-e9b70e99711d6d9bd879922b2917fb816ea2149a235f2386980dd96707ba3bea in particular).

Once that PR lands, it should become easier to implement the change suggested here. (Or feel free to add a similarly named file right away.)

Reply:
I'll plan to implement the change here as a follow up to the tagged PR, thanks for the context!